frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
+          <MetricsSwitcher artifacts={artifacts} artifactTypes={artifactTypes}></MetricsSwitcher>
+        )}
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {classificationMetricsArtifacts &&
+        classificationMetricsArtifacts.length > 0 &&
+        classificationMetricsArtifacts.map(artifact => {
+          const id = artifact.getId();
+          const customProperties = artifact.getCustomPropertiesMap();
+          const confidenceMetrics = customProperties
+            .get('confidenceMetrics')
+            ?.getStructValue()
+            ?.toJavaScript();
+          return (
+            <React.Fragment key={id}>
+              {confidenceMetrics && (
+                <div>
+                  <div className={padding(40, 'lr')}>
+                    <h1>
+                      ROC Curve{' '}
+                      <IconWithTooltip
+                        Icon={HelpIcon}
+                        iconColor={color.weak}
+                        tooltip={ROC_CURVE_DEFINITION}
+                      ></IconWithTooltip>
+                    </h1>
+                  </div>
+                  <ROCCurve configs={buildRocCurveConfig((confidenceMetrics as any).list)} />
+                </div>
+              )}
+            </React.Fragment>
+          );
+        })}
+    </>
+  );
+}
+
+function buildRocCurveConfig(confidenceMetricsArray: ConfidenceMetrics[]): ROCCurveConfig[] {
+  let displayPoint: DisplayPoint[] = [];
+  for (let metrics of confidenceMetricsArray) {
+    displayPoint.push({
+      label: metrics.confidenceThreshold,
+      x: metrics.falsePositiveRate,
+      y: metrics.recall,
+    });
+  }
```

Comment:
nit: also can be simplified as (in typescript, this code base typically prefer using map, filter style code.

```
return [
  {
    data: confidenceMetrics.map(metric => ({label: metric.confidenceThreshold, x: metric.falsePositiveRate, y: metric.recall})),
  },
];
```

Why do we need the type force casting `as ROCCurveConfig[]`?

Reply:
Thank you Yuan! Very good simplification here. 

The reason of specifying ROCCurveConfig is that `<ROCCurve>` component requires input being typed. Otherwise the error is

```
Property 'type' is missing in type '{ data: any; }' but required in type 'ROCCurveConfig'.ts(2741)
```

Reply:
After discussion: we should add type because this `ROCCurveConfig` extends a parent class.